chain/chain/src/doomslug.rs

```diff
@@ -420,25 +426,27 @@ impl Doomslug {
     pub fn can_approved_block_be_produced(
         mode: DoomslugThresholdMode,
         approvals: &[Option<Signature>],
-        stakes: &Vec<(Balance, Balance)>,
+        stakes: &Vec<(Balance, Balance, bool)>,
     ) -> bool {
         if mode == DoomslugThresholdMode::NoApprovals {
             return true;
         }
 
-        let threshold1 = stakes.iter().map(|(x, _)| x).sum::<Balance>() * 2 / 3;
-        let threshold2 = stakes.iter().map(|(_, x)| x).sum::<Balance>() * 2 / 3;
+        let threshold1 = stakes.iter().map(|(x, _, _)| x).sum::<Balance>() * 2 / 3;
+        let threshold2 = stakes.iter().map(|(_, x, _)| x).sum::<Balance>() * 2 / 3;
 
         let approved_stake1 = approvals
             .iter()
             .zip(stakes.iter())
-            .map(|(approval, (stake, _))| if approval.is_some() { *stake } else { 0 })
+            .filter(|(_, (_, _, is_slashed))| !*is_slashed)
```

Comment:
might be better to use `filter_map` directly https://nnethercote.github.io/perf-book/iterators.html#chaining